test/e2e/azure_test.go

```diff
@@ -122,6 +122,17 @@ var _ = Describe("Workoad cluster creation", func() {
 				WaitForControlPlaneIntervals: e2eConfig.GetIntervals(specName, "wait-control-plane"),
 				WaitForMachineDeployments:    e2eConfig.GetIntervals(specName, "wait-worker-nodes"),
 			})
+
+			Context("in a cluster with one or more nodes", func() {
```

Comment:
Does this new spec add a significant duration to the test? if not, let's add it to the single control plane cluster as well

Reply:
It doesn't add a lot of time (about two minutes max?), but even after adding the right tolerations, I wasn't able to get the test to run on the single control plane cluster. The error was something about "LB can't be created" that implied it needs worker nodes.

If that rings a bell or you have an idea what might be going wrong there, I would love to fix this up so it runs in both cluster configurations.

Reply:
I see. We should consider adding a worker node or two to the single CP test, a cluster with a control plane and no workers isn't really a useful scenario to cover

Reply:
unless you plan to make this update in this PR, its lgtm

Reply:
I'm good with doing that as a follow up PR too, this is already a good addition as-is

Reply:
I can do that change in a separate PR, so this one can proceed.